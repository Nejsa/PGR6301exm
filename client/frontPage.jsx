import React from "react";
import { Link } from "react-router-dom";

function FrontPage() {
    return (
        <div>
            <h1>Home page for Activities</h1>
            <ul>
                <li>
                    <Link to="/logHours">Log your hours</Link>
                </li>
                <li>
                    <Link to="/ShowActivities">Show Available Activities</Link>
                </li>
            </ul>
        </div>
    );
}

export default FrontPage;