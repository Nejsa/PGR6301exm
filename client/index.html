<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Time Management app</title>
</head>
<body>
<div id="app">
</div>
</body>
<script src="index.jsx" type="module"></script>
</html>