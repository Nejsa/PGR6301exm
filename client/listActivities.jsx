import React from "react";
import { Link } from "react-router-dom";
import { fetchJSON, useLoader } from "./index";

function ListActivities() {
    // Dummy data no longer in use cause of MongoDB data
    const activities = [
        { activity: "Refactor Code", available: true },
        { activity: "Finish Frontend Code", available: true },
        { activity: "Finish Backend Code", available: true },
    ];

    const { loading, error, data } = useLoader(async () => {
        return fetchJSON("/api/activities")
    })

    const availableActivities = activities.filter((activity) => activity.available);
    console.log(data)
    if (loading) {``
        return <div> Still Loading... </div>;
    }

    if (error) {
        return (
            <div>
                <h1> Error </h1>
                <div> {error.toString()} </div>
            </div>
        );
    }

    return (
        <div>
            <h1>Available Activities:</h1>
            <div>
                <Link to="/">Go back to home</Link>
            </div>
            <p>Number of Available Activities: {data?.length}</p>
            <ul>
                {data?.map((activity) => (
                    <li key={activity.activity}>{activity.activity}</li>



                ))}
            </ul>
        </div>
    );
}




export default ListActivities;