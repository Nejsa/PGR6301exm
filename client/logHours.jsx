import React from "react";
import {Link} from "react-router-dom";



function LogHours() {
    return (
        <div>
            <div>
                <h1> Log your Hours </h1>
                <Link to={"/"}> Go to Activities </Link>
            </div>
            <div>
                <Link to="/">Go back to home</Link>
            </div>
        </div>
    );
}

export default LogHours;
