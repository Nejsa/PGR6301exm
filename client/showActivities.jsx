import React from "react";
import {Link} from "react-router-dom";

function ShowActivities() {
    return (
        <div>
            <div>
            <h1> Show Activities Available </h1>
                <Link to={"/logHours"}> Log your hours </Link>
            </div>
            <div>
                <Link to={"/"}> Go back to home </Link>
            </div>
        </div>
    );
}

export default ShowActivities;
