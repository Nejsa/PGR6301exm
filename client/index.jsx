import React, {useState, useEffect} from "react";
import { createRoot } from "react-dom/client";
import {BrowserRouter, json, Link, Route, Routes} from "react-router-dom";
import LogHours from "./logHours";


const element = document.getElementById("app");
const root = createRoot(element);


function FrontPage() {
    const { loading, error, data } = useLoader(async () => {
    return fetchJSON("/api/activities");
});


    if(loading) {
        return <div> Still Loading... </div>
    }

    if(error) {
        return (
            <div>
                <h1> Error </h1>
                <div> {error.toString()} </div>
            </div>
        );
    }

    return(
        <div>
            <h1> Home page for Activities</h1>
            <p>Its currently available: {
                data.length
            }</p>
            <ul>
                <li>
                    <Link to={"/logHours"}> Log your hours </Link>
                </li>
                <li>
                    <Link to={"/ShowActivities"}> Show Available Activities </Link>
                </li>
            </ul>
        </div>
    );
}

async function fetchJSON(url) {
    const res = await fetch(url)
    if (!res.ok){
        throw new Error (`fail to load${res.status}`)
    }
    return await res.json()

}

function useLoader(loadingFunction){
    const [loading, setLoading] = useState(true);
    const [error, setError] = useState();
    const [data, setData] = useState();

    async function load(){
        try{
            setLoading(true);
            setData(await loadingFunction());
        }
        catch (error){
            setError(error);
        }
        finally {
            setLoading(false);
        }
    }

    useEffect(() => load(), []);

    return { loading, error, data };
}



export function ShowActivities(ListActivities){
    const { loading, error, data } = useLoader(async () => {
        return fetchJSON("/api/activities");
    });


    if(loading) {
        return <div> Still Loading... </div>
    }

    if(error) {
        return (
            <div>
                <h1> Error </h1>
                <div> {error.toString()} </div>
            </div>
        );
    }

    return (
        <div>
            <h1> Available Activities: </h1>
            <div>
            <div>
                <Link to={"/"}> Go back to home </Link>
            </div>
            <div>
                <Link to={"/logHours"}> Log your hours </Link>
            </div>
        </div>
            {data.map((activity) => (
                <div key={activity.hours}>
                    <h1> {activity.activity} </h1>
                    <p> Hours: {activity.hours} </p>
                </div>
            ))}
        </div>
    );
}



    async function handleSumbit(e){
        e.preventDefault()

        await fetchJSON("/api/activities", {
            method: "post",
            json: {hours, activity}
        })

        setHours("")
        setActivity("")

}


function Application() {
   async function ListActivities(){
        return await fetchJSON("api/activities")

    }
    return (
        <BrowserRouter>
            <Routes>
                <Route path={"/"} element={<FrontPage ListActivities={ListActivities} />} />
                <Route path={"/ShowActivities/*"} element={<ShowActivities ListActivities={ListActivities} />} />
                <Route path={"/logHours/*"} element={<LogHours />} />
            </Routes>
        </BrowserRouter>
    );
}

// root.render(<h1>Something Displayed!</h1>)


root.render(<Application />);





