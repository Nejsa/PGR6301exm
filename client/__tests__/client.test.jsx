import * as React from "react";
import { createRoot } from "react-dom/client";
import FrontPage from "../frontPage.jsx";



describe('client test suite', () => {
    it('frontPage gets rendered', () => {
        const element = document.createElement("div")


        const root = createRoot(element)


        root.render(<FrontPage />)

        expect(element.querySelector("h1"))
        expect(element.innerHTML).toMatchSnapshot();
    });

});