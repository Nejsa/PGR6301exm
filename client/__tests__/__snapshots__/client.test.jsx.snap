// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client test suite frontPage gets rendered 1`] = `""`;
